client/asset/btc/electrum.go

```diff
@@ -130,16 +134,38 @@ func (btc *ExchangeWalletElectrum) Connect(ctx context.Context) (*sync.WaitGroup
 			genesis.String(), serverFeats.Genesis)
 	}
 
-	dbWG, err := btc.startTxHistoryDB(ctx)
+	verStr, err := btc.ew.wallet.Version(ctx)
+	if err != nil {
+		return nil, err
+	}
+	gotVer, err := dex.SemverFromString(verStr)
 	if err != nil {
 		return nil, err
 	}
+	if !dex.SemverCompatible(btc.minElectrumVersion, *gotVer) {
+		return nil, fmt.Errorf("wanted electrum wallet version %s but got %s", btc.minElectrumVersion, gotVer)
+	}
 
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		dbWG.Wait()
-	}()
+	if btc.minElectrumVersion.Major >= 4 && btc.minElectrumVersion.Major >= 5 {
```

Comment:
Doesn't the second condition imply the first? And is this version correct for all assets?

Reply:
Sorry that second one should be minor... did I even test this.

Only btc has made it to this version so far. 4.5.x

Reply:
Changed the second Major to Minor. Sorry I forgot to push the rebase, but that was the only change.